### PR TITLE
fix(connector): use bound channel for source

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

As title, fix OOM

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]

File: src/source/src/connector_source.rs [hunk 1]
```diff
@@ -263,7 +263,7 @@ impl ConnectorSource {
         splits: ConnectorState,
         column_ids: Vec<ColumnId>,
     ) -> Result<ConnectorSourceReader> {
-        let (tx, rx) = mpsc::unbounded_channel();
+        let (tx, rx) = mpsc::channel(CONNECTOR_MESSAGE_BUFFER_SIZE);
```
Comment (shanicky) [inline src/source/src/connector_source.rs:266]: Would it be better to set the number of splits instead of 64

**Review by shanicky — APPROVED** [review]
rest LGTM

File: src/source/src/connector_source.rs (hunk 1 above)
Comment (skyzh) [reply src/source/src/connector_source.rs:266]: Is there any reason behind that?

Comment (shanicky) [reply src/source/src/connector_source.rs:266]: 64 should be okay, just ignore me

Comment (skyzh) [reply src/source/src/connector_source.rs:266]: It's too small 🤤🤤 still benchmarking. I'm guessing it should be 16 * chunk size.

**Pull request merged**